ui/src/app/pages/analysis/runtime-panel.tsx

```diff
@@ -499,7 +609,12 @@ export const RuntimePanel = fp.flow(
   const initialMasterMachine = findMachineByName(machineName) || defaultMachineType;
   const initialCompute = dataprocConfig ? ComputeType.Dataproc : ComputeType.Standard;
   // TODO(RW-5591): Initialize PanelContent according to the runtime status.
```

Comment:
So initially I thought this might work, but I'm less sure as I understand more about hooks. My updated understanding is that this only works if the runtime store has _already_ been populated by the time the user renders the panel contents. In most cases, this will be true, but I don't think it's guaranteed.

Reply:
Sorry, what do you mean by 'this'? Inspecting the runtime status?

Reply:
The thing you did here and which is outlined in this TODO, which is to pick `initialPanelContent` based on the incoming runtime. The initial value of a hook is only relevant to the first time this component gets rendered - if the runtime is undefined at that point (still loading), that dictates the initial value.

Reply:
Given the inability to rely on `currentRuntime`, do you think it's worth trying to do another useEffect that updates the panel content when the runtime status is updated, or should I just set reasonable defaults based on the initial state of `currentRuntime` and let the user click through to the customize page if they need to?

Reply:
It's probably easiest to just show the Customize page in this case as the initial page, if it's still loading. Showing the create page will be confusing if you already have an active runtime.

The useEffect approach I think is also possible, but would be complicated since you'll have to make sure it would run only once. Probably you'd need to introduce some loading state or PanelContent.Loading type, then useEffect transition based on runtime becoming !== undefined.

I would just go with the simple approach of defaulting to Customize for now and leave a comment explaining the current state of things (please also remove this TODO line). We can revisit if this race condition winds up being more common than I'm expecting it to be.

Reply:
I think that would result in this Create page never being shown - we can't currently differentiate between `getRuntime` having not yet returned and `getRuntime` having returned nothing.

Reply:
Ok, I think I am slightly wrong, and that `currentRuntime` will be `undefined` when `getRuntime` has not returned for the first time and it will be `null` if `getRuntime` 404s.